lib/Analysis/Allocation.cpp

```diff
@@ -374,35 +440,81 @@ class AllocationAnalysis {
     });
   }
 
+  void updateBufferRange(BufferT *buffer, IntervalT interval) {
+    if (bufferRange.contains(buffer))
+      interval = interval.merge(bufferRange[buffer]);
+    bufferRange[buffer] = interval;
+    LLVM_DEBUG(llvm::dbgs() << "    UPDATE BUFFER: " << buffer->id << " -> "
+                            << interval << "\n");
+  }
+
   /// Computes the liveness range of the allocated value.
   /// Each buffer is allocated only once.
-  void resolveExplicitBufferLiveness(
-      function_ref<Interval<size_t>(Value value)> getLiveness) {
+  void resolveExplicitBufferLiveness(LivenessF getLiveness) {
     for (auto valueBufferIter : allocation->valueBuffer) {
       auto value = valueBufferIter.first;
       auto *buffer = valueBufferIter.second;
-      bufferRange[buffer] = getLiveness(value);
+      auto ranges = getLiveness(value);
+      // TODO(SJW): bufferRange should support disjoint intervals
```

Comment:
Can you clarify the relationship between the code and the comment?

Reply:
Yeah, my concern was actually that it could be explained in the liveness chart you plotted. It might be difficult for others to connect the `updateBufferRange` function with what you said here. Personally, I think you can remove the comment as long as it's super clear in the comment below.